syntax = "proto3";

package lb;


//from services to LB

message InitRequest{
    string service_name = 1;
    string instance_id = 2;
    string session_token = 3; 
}

message InitResponse{
    string session_token = 1;
    bool accepted = 2;
    string reason = 3;
}

message KeepAlive{
    string session_token = 1;
    uint64 timestamp = 2;
}

message ServiceReport{
    string session_token = 1;
    uint32 cpu_usage = 2;
    uint32 memory_usage = 3;
    uint32 active_requests = 4;
}

message CloseRequest{
    string service_id = 1;
    string reason = 2;
}


//from LB to services

message GetReport{
    string session_token = 1; 
}

message GetReportAck{
    bool ok = 1;
    string message = 2;
}

message CloseAck{
    bool ok = 1;
    string message = 2;
}


